cpp/src/io/orc/stripe_init.cu

```diff
@@ -226,6 +226,26 @@ enum row_entry_state_e {
   STORE_INDEX2,
 };
 
+/**
+ * @brief Calculates the order of index streams based on the index types present in the column.
+ *
+ * @param index_types_bitmap The bitmap of index types showing which index streams are present
+ *
+ * @return The order of index streams
+ */
+static auto __device__ index_order_from_index_types(uint32_t index_types_bitmap)
```

Comment:
Technically all free functions are static.

Reply:
Maybe I'm misunderstanding the comment, but `static` is used here to give the function internal linkage - it is not called outside of this translation unit. Maybe this is not relevant for `__device__` functions, but we use this extensively in cuIO.

Reply:
Perhaps @ttnghia is confusing the concept of [static member variables](https://en.cppreference.com/w/cpp/language/static) with usage of `static` as a [storage class specifier](https://en.cppreference.com/w/cpp/language/storage_duration)? They are two completely different concepts that unfortunately reuse the same keyword. Otherwise I am not sure what the comment is referring to.

Reply:
> but static is used here to give the function internal linkage

This makes sense. Otherwise, I don't see why it needs to be `static`. Sorry for the confusion.

Typically we can just put the function into an anonymous namespace for that purpose.